iree/vm/test/list_ops.mlir

```diff
@@ -69,4 +90,27 @@ vm.module @list_ops {
     vm.return
   }
 
+  //===--------------------------------------------------------------------===//
+  // vm.list.* with control flow
+  //===--------------------------------------------------------------------===//
+
+  vm.export @test_control_flow
+  vm.func @test_control_flow() {
+    // TODO(simon-camp): Implement type converter in the Vm to EmitC conversion. 
+  //   %c0 = vm.const.i32 0 : i32
+  //   %c1 = vm.const.i32 1 : i32
+  //   %c2 = vm.const.i32 2 : i32
+  //   %c1_dno = iree.do_not_optimize(%c1) : i32
+  //   %list = vm.list.alloc %c2 : (i32) -> !vm.list<i32>
+  //   vm.list.resize %list, %c2 : (!vm.list<i32>, i32)
+  //   vm.list.set.i32 %list, %c0, %c1 : (!vm.list<i32>, i32, i32)
+  //   vm.list.set.i32 %list, %c1, %c2 : (!vm.list<i32>, i32, i32)
+  //   vm.cond_br %c1_dno, ^bb1(%list : !vm.list<i32>), ^bb2
+  // ^bb1(%list_0 : !vm.list<i32>):
```

Comment:
is the block argument being retained here as part of the branch? branches - especially cond ones where there are two different paths - require some smarts (is the value retained by both successors, just one, or none, etc) and some more liveness checks (same ref instance passed as multiple branch arguments, etc).

(you may already be doing this, but I'd thought there would be some changes in this PR too)

Reply:
I suspected this would need special handling (not implemented yet), so I wanted to test this. I didn't manage to write a type converter for the block arguments yet. (We are going from `!vm.list<...>` to `!emitc.opaque<"iree_vm_ref_t">`)

I can delete the test for now and add it back in a follow up.

Reply:
up to you - fine to have it in if it's useful to you :)